## Fun stuffs

In [18]:
from sklearn.neighbors  import KNeighborsClassifier
#Prerequistite
from sklearn.feature_extraction.text  import CountVectorizer
import numpy as np
import pandas as pd
lang_of_girls=['mmmmmm i hate u a lot(:(:','kkkkk i hate hate u','bye imm gonna block you''']

'''Learn the vocabulary of girls ,pretty hard though(:'''
vector=CountVectorizer()
vector.fit(lang_of_girls)

document_mat=vector.transform(lang_of_girls)
'''Transform the vector'''
array=document_mat.toarray()
print(vector.get_feature_names())
file=pd.DataFrame(array,columns=vector.get_feature_names())
test=['I hate u a lot']
test_dtm=vector.transform(test)


d= pd.DataFrame(test_dtm.toarray(),columns=vector.get_feature_names())
d


    

['block', 'bye', 'gonna', 'hate', 'imm', 'kkkkk', 'lot', 'mmmmmm', 'you']


block  bye  gonna  hate  imm  kkkkk  lot  mmmmmm  you
0      0    0      0     1    0      0    1       0    0

## The actual stuffs

In [19]:
import pandas as pd
file='sms.tsv'
sms=pd.read_table(file,header=None,names=['label','message'])

In [20]:
#Examine the class distribution
sms.label.values

array(['ham', 'ham', 'spam', ..., 'ham', 'ham', 'ham'], dtype=object)

  ## Finding how many values are there in the both classes

In [21]:

sms.label.value_counts()

ham     4825
spam     747
Name: label, dtype: int64

## Add a new column that maps ham to 0 and spam to 1 for easy computation

In [22]:
sms['mapped']=sms.label.map({'ham':0,'spam':1})

In [23]:
sms.head(10)

label                                            message  mapped
0   ham  Go until jurong point, crazy.. Available only ...       0
1   ham                      Ok lar... Joking wif u oni...       0
2  spam  Free entry in 2 a wkly comp to win FA Cup fina...       1
3   ham  U dun say so early hor... U c already then say...       0
4   ham  Nah I don't think he goes to usf, he lives aro...       0
5  spam  FreeMsg Hey there darling it's been 3 week's n...       1
6   ham  Even my brother is not like to speak with me. ...       0
7   ham  As per your request 'Melle Melle (Oru Minnamin...       0
8  spam  WINNER!! As a valued network customer you have...       1
9  spam  Had your mobile 11 months or more? U R entitle...       1

## Training the model

In [1]:
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.cross_validation import train_test_split
vectorizer=CountVectorizer()








C:\Anaconda\lib\site-packages\sklearn\cross_validation.py:41: DeprecationWarning: This module was deprecated in version 0.18 in favor of the model_selection module into which all the refactored classes and functions are moved. Also note that the interface of the new CV iterators are different from that of this module. This module will be removed in 0.20.
  "This module will be removed in 0.20.", DeprecationWarning)


## Equally combining fit and transform






In [31]:
dtm=vectorizer.fit_transform(x_train)

In [32]:
dtm.shape


(4179, 7456)

## In this 4179 represents the number of rows (ie ,words) and 7456 represents what the model has learnt (i.e) induvidual words etc..

## Using Naive Bayes Classification model

